presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -234,6 +234,17 @@ private boolean tryFetchLookupSourceProvider()
             }
             lookupSourceProvider = requireNonNull(getDone(lookupSourceProviderFuture));
             statisticsCounter.updateLookupSourcePositions(lookupSourceProvider.withLease(lookupSourceLease -> lookupSourceLease.getLookupSource().getJoinPositionCount()));
+
+            // check if we can finish the join earlier
+            if (!probeOnOuterSide &&
+                    !finishing &&
+                    !finished &&
+                    lookupSourceProvider.withLease(lookupSourceLease ->
+                            // If the lookup has spilled, return true as it is assumed there were rows.
+                            lookupSourceLease.hasSpilled() ? false : lookupSourceLease.getLookupSource().isEmpty())) {
```

Comment:
Why you need this `?:`?
spilled lookup source is not empty, right? So just `lookupSourceLease.getLookupSource().isEmpty()` should be equivalent.

(Then also remove the preceding comment.)